### ISSUE TITLE
Bug: [return-await] should handle nested try-catch-finally blocks with "in-try-catch" option

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#8907]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.2.2&showAST=es&fileType=.tsx&code=IYZwngdgxgBAZgV2gFwJYHsLwBQEoYDeAUDDMgE5iEmlmXW0wC%2B8qEwANh1cY6QPT8YwOHACmUZCBhjy5dORgALYBAAmHNgHMY6OLoTJZMKMGRQlNRuTHIE5LAAV5AW1QgxAOhsArCcjwAbitmGhZTcyVqMKImIiJQSFhEFAwsOAAmPAY6HjCTMwsc0go8xhY4Nk5uYtpBYVF-aVl5RRV1TQgdPQMjRUr2LjAQ0hs7BxhndDcPbzE-SSCQuNIKqqHamKYgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aRfWaJ2gQwDufSvnRgA2pETRoAe2iQANFGa180QrTJ98ZABaQAuuDABfEKaA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
async function f() {
  try {
    try {
    } finally {
      // affects error handling of outer catch
      return Promise.reject();
    }
  } catch {

  }
}

// or,
async function f2() {
  try {
  } catch {
    try {
    } finally {
      // affects error handling of outer finally
      return Promise.reject();
    }
  } finally {
    
  }
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/return-await": ["error", "in-try-catch"]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expect that the commented promises should flag as requiring await.

### Actual Result

no errors

### Additional Info

The `in-try-catch` deductions are generally buggy. Note that putting an `await` doesn't cause the rule to flag that it's unnecessary either, despite [the rule's documentation implying that there will only be one correct choice in each scenario](https://typescript-eslint.io/rules/return-await/#in-try-catch).

 See also https://github.com/typescript-eslint/typescript-eslint/issues/8663

Note that putting a `return` in a `catch` suffers the same defect and is probably a more realistic example for this issue.
